src/impl.cpp

```diff
@@ -1 +1,19 @@
 #include "impl.h"
+int BinaryInvert(int x,int p,int n)
+{
+    unsigned int mask {x}, mask2 {~x};
```

Comment:
we can complete this task with just one mask, filled by true bits
too complicated solution